src/Sulu/Component/Content/Import/Format/Xliff12.php

```diff
@@ -0,0 +1,202 @@
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function parse($filePath, $locale)
+    {
+        $dom = XmlUtils::loadFile($filePath);
+
+        return $this->extractData($dom, $locale);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getProperty($name, $data, $contentTypeName = null, $extension = null, $default = null)
+    {
+        $propertyName = '';
+
+        if ($extension) {
+            $propertyName = $extension . '-';
+        }
+
+        $propertyName .= $name;
+
+        if (!isset($data[$propertyName])) {
+            return $default;
+        }
+
+        $property = $data[$propertyName];
+        $property['type'] = $contentTypeName;
+
+        return $property;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPropertyData($name, $data, $contentTypeName = null, $extension = null, $default = null)
+    {
+        $property = $this->getProperty($name, $data, $contentTypeName, $extension);
+
+        if (isset($property['children'])) {
+            $data = [];
+            foreach ($property['children'] as $key => $child) {
+                $childProperties = [];
+                foreach (array_keys($child) as $childKey) {
+                    $childProperties[$childKey] = $this->getPropertyData($childKey, $child, $contentTypeName = null, $extension = null);
+                }
+                $data[$key] = $childProperties;
+            }
+
+            return $data;
+        }
+
+        if (!isset($property['value'])) {
+            return $default;
+        }
+
+        return $property['value'];
+    }
+
+    /**
+     * @param \DOMDocument $dom
+     * @param string $locale
+     *
+     * @return array
+     */
+    protected function extractData(\DOMDocument $dom, $locale)
+    {
+        $xml = simplexml_import_dom($dom);
+        $encoding = strtoupper($dom->encoding);
+        $xml->registerXPathNamespace('xliff', 'urn:oasis:names:tc:xliff:document:1.2');
+
+        $documents = [];
+
+        foreach ($xml->xpath('//xliff:file') as $file) {
+            $fileAttributes = $file->attributes();
+
+            if (!isset($fileAttributes['original'])) {
+                continue;
+            }
+
+            $uuid = (string) $fileAttributes['original'];
+            $data = $this->getData($file, $encoding);
+
+            $template = null;
+
+            if (isset($data['structureType'])) {
+                $template = $data['structureType']['value'];
+                unset($data['structureType']);
+            }
+
+            $documents[] = [
+                'uuid' => $uuid,
+                'locale' => $locale,
+                'structureType' => $template,
+                'data' => $data,
+            ];
+        }
+
+        return $documents;
+    }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    protected function getData($file, $encoding)
+    {
+        $data = [];
+
+        foreach ($file->body->children() as $translation) {
```

Comment:
dont nest loops and ifs ... also here own methods and early returns makes it much more readable.